File: examserver/src/main/java/com/example/examserver/service/QuizService.java
package com.example.examserver.service;

import java.util.List;
import java.util.Set;

import com.example.examserver.model.exam.Category;
import com.example.examserver.model.exam.Quiz;


public interface QuizService {
	public Quiz addQuiz(Quiz quiz);
	public Quiz updateQuiz(Quiz quiz);
	public Set<Quiz> getQuizzes();
	
	public Quiz getQuiz(Long quizId);
	
	public void deleteQuiz(Long quizId);
	
	public List<Quiz> getQuizzesOfcategory(Category category);
}

File: ExamPortalUI/src/app/services/quiz.service.ts
import { Injectable } from '@angular/core';
import BASEURL from './helper';
import { HttpClient } from '@angular/common/http';

@Injectable({
  providedIn: 'root'
})
export class QuizService {
  constructor(private http: HttpClient) { }

  public quizzes() {
    return this.http.get(`${BASEURL}/quiz/`)
  }

  public addQuiz(quiz: any) {
    return this.http.post(`${BASEURL}/quiz/`, quiz)
  }
  public deleteQuiz(qid:any){
    return this.http.delete(`${BASEURL}/quiz/${qid}`)
  }
  public getQuiz(qid:any){
    return this.http.get(`${BASEURL}/quiz/${qid}`)
  }
  public updateQuiz(quiz:any){
    return this.http.put(`${BASEURL}/quiz/`, quiz)
  }
  public geQuizzesOfcategory(cid:any){
    return this.http.get(`${BASEURL}/quiz/category/${cid}`)
  }
}
